common/src/main/java/com/wynntils/core/features/Feature.java

```diff
@@ -30,20 +36,39 @@
 public abstract class Feature {
     private ImmutableList<Condition> conditions;
     private boolean isListener = false;
+    private boolean isOverlayHolder = false;
     private List<KeyHolder> keyMappings = new ArrayList<>();
     private List<ConfigHolder> configOptions = new ArrayList<>();
+    private List<Overlay> overlays = new ArrayList<>();
 
     protected boolean enabled = false;
 
     public final void init() {
         ImmutableList.Builder<Condition> conditions = new ImmutableList.Builder<>();
 
+        if (this.isOverlayHolder) initOverlays();
+
         onInit(conditions);
 
         this.conditions = conditions.build();
 
-        if (this.conditions.isEmpty()) return;
-        this.conditions.forEach(Condition::init);
+        if (!this.conditions.isEmpty()) this.conditions.forEach(Condition::init);
+    }
+
+    private void initOverlays() {
+        Field[] overlayFields = FieldUtils.getFieldsWithAnnotation(this.getClass(), OverlayInfo.class);
+        for (Field overlayField : overlayFields) {
+            if (overlayField.getType() != Overlay.class) continue;
```

Comment:
Actually, I realized you also need to look for `List.class` here, to be able to support the use case of multiple overlays, and iterate over the elements of that list, adding the overlays. We can do that later on as well, so long as we're not forgetting it.

Reply:
List support is not currently added, I will work on it separately.